lib/gui/models/flash-state.js

```diff
@@ -20,212 +20,203 @@
-  };
-
-  /**
-   * @summary Get last flash source checksum
-   * @function
-   * @public
-   *
-   * @description
-   * This function returns undefined if there was no last flash.
-   *
-   * @returns {(String|Undefined)} the last flash source checksum
-   *
-   * @example
-   * const checksum = FlashStateModel.getLastFlashSourceChecksum();
-   */
-  this.getLastFlashSourceChecksum = () => {
-    return this.getFlashResults().sourceChecksum;
-  };
-
-  /**
-   * @summary Get last flash error code
-   * @function
-   * @public
-   *
-   * @description
-   * This function returns undefined if there was no last flash.
-   *
-   * @returns {(String|Undefined)} the last flash error code
-   *
-   * @example
-   * const errorCode = FlashStateModel.getLastFlashErrorCode();
-   */
-  this.getLastFlashErrorCode = () => {
-    return this.getFlashResults().errorCode;
-  };
-
-});
-
-module.exports = MODULE_NAME;
+/**
+ * @summary Reset flash state
+ * @function
+ * @public
+ *
+ * @example
+ * flashState.resetState();
+ */
+exports.resetState = () => {
+  Store.dispatch({
+    type: Store.Actions.RESET_FLASH_STATE
+  });
+};
+
+/**
+ * @summary Check if currently flashing
+ * @function
+ * @private
+ *
+ * @returns {Boolean} whether is flashing or not
+ *
+ * @example
+ * if (flashState.isFlashing()) {
+ *   console.log('We\'re currently flashing');
+ * }
+ */
+exports.isFlashing = () => {
+  return Store.getState().toJS().isFlashing;
+};
+
+/**
+ * @summary Set the flashing flag
+ * @function
+ * @private
+ *
+ * @description
+ * This function is extracted for testing purposes.
+ *
+ * The flag is used to signify that we're going to
+ * start a flash process.
+ *
+ * @example
+ * flashState.setFlashingFlag();
+ */
+exports.setFlashingFlag = () => {
+  Store.dispatch({
+    type: Store.Actions.SET_FLASHING_FLAG
+  });
+};
+
+/**
+ * @summary Unset the flashing flag
+ * @function
+ * @private
+ *
+ * @description
+ * This function is extracted for testing purposes.
+ *
+ * The flag is used to signify that the write process ended.
+ *
+ * @param {Object} results - flash results
+ *
+ * @example
+ * flashState.unsetFlashingFlag({
+ *   cancelled: false,
+ *   sourceChecksum: 'a1b45d'
+ * });
+ */
+exports.unsetFlashingFlag = (results) => {
+  Store.dispatch({
+    type: Store.Actions.UNSET_FLASHING_FLAG,
+    data: results
+  });
+};
+
+/**
+ * @summary Set the flashing state
+ * @function
+ * @private
+ *
+ * @description
+ * This function is extracted for testing purposes.
+ *
+ * @param {Object} state - flashing state
+ *
+ * @example
+ * flashState.setProgressState({
+ *   type: 'write',
+ *   percentage: 50,
+ *   eta: 15,
+ *   speed: 100000000000
+ * });
+ */
+exports.setProgressState = (state) => {
+  Store.dispatch({
+    type: Store.Actions.SET_FLASH_STATE,
+    data: {
+      type: state.type,
+      percentage: state.percentage,
+      eta: state.eta,
+
+      speed: _.attempt(() => {
```

Comment:
I know I've mentioned this before, so apologies for sounding like a broken record, but my gut feeling is still that this conversion from bytes to Megabytes 'belongs' in the GUI itself, rather than in the internal model.
(but there may well be other aspects to this that I'm not aware of)